WordPress/openverse
# Convert longer media `varchar` fields to `text` in the catalog db

## Fixes



Fixes #4312 by @AetherUnbound

## Description



This PR alters the SQL DDL files for the catalog database to reflect the changes that will be made to the existing catalog database with the following migration:

```sql
-- Alter image table
ALTER TABLE image ALTER COLUMN title TYPE TEXT USING title :: TEXT;
ALTER TABLE image ALTER COLUMN foreign_identifier TYPE TEXT USING foreign_identifier :: TEXT;
ALTER TABLE image ALTER COLUMN foreign_landing_url TYPE TEXT USING foreign_landing_url :: TEXT;
ALTER TABLE image ALTER COLUMN url TYPE TEXT USING url :: TEXT;
ALTER TABLE image ALTER COLUMN thumbnail TYPE TEXT USING thumbnail :: TEXT;
ALTER TABLE image ALTER COLUMN creator TYPE TEXT USING creator :: TEXT;
ALTER TABLE image ALTER COLUMN creator_url TYPE TEXT USING creator_url :: TEXT;

-- audioset_view must be dropped and recreated
-- THe view is also dependent on audio_set_foreign_identifier so we need to drop it before altering the base table
DROP VIEW IF EXISTS audioset_view;

-- Alter audio table
ALTER TABLE audio ALTER COLUMN title TYPE TEXT USING title :: TEXT;
ALTER TABLE audio ALTER COLUMN foreign_identifier TYPE TEXT USING foreign_identifier :: TEXT;
ALTER TABLE audio ALTER COLUMN foreign_landing_url TYPE TEXT USING foreign_landing_url :: TEXT;
ALTER TABLE audio ALTER COLUMN url TYPE TEXT USING url :: TEXT;
ALTER TABLE audio ALTER COLUMN thumbnail TYPE TEXT USING thumbnail :: TEXT;
ALTER TABLE audio ALTER COLUMN creator TYPE TEXT USING creator :: TEXT;
ALTER TABLE audio ALTER COLUMN creator_url TYPE TEXT USING creator_url :: TEXT;
ALTER TABLE audio ALTER COLUMN audio_set_foreign_identifier TYPE TEXT USING audio_set_foreign_identifier :: TEXT;

-- Recreate the audioset_view 
CREATE VIEW audioset_view AS
  SELECT DISTINCT ON (audio.audio_set_foreign_identifier, audio.provider)
    audio.audio_set_foreign_identifier          ::text AS foreign_identifier,
    audio.audio_set ->> 'title'                 ::text AS title,
    audio.audio_set ->> 'foreign_landing_url'   ::text AS foreign_landing_url,
    audio.audio_set ->> 'creator'               ::text AS creator,
    audio.audio_set ->> 'creator_url'           ::text AS creator_url,
    audio.audio_set ->> 'url'                   ::text AS url,
    (audio.audio_set ->> 'filesize'::text)      ::integer AS filesize,
    (audio.audio_set ->> 'filetype'::text)      ::character varying(80) AS filetype,
    audio.audio_set ->> 'thumbnail'             ::text AS thumbnail,
    audio.provider
FROM audio
WHERE (audio.audio_set_foreign_identifier IS NOT NULL AND audio.audio_set IS NOT NULL)
ORDER BY
    audio.audio_set_foreign_identifier,
    audio.provider,
    audio.updated_on DESC;
```

Once the above migration is reviewed and approved, we can run it on the catalog database and merge this PR.

I've also updated some of the mock schemas for the ingestion server testing. We won't need these in perpetuity, fortunately, thanks to #3925.


## Testing Instructions


1. Checkout `main`
1. Run `just down -v && just api/init`
1. Check out this branch
1. Run `just catalog/pgcli`, then the above migrations in order (you'll need to copy out each block below the comments, pgcli doesn't seem to play nicely with comments)
1. Run `\d image`, `\d audio`, and `\d audioset_view` in the shell to verify the tables are now using `text`
1. Run `just c` to start the catalog and start the Jamendo and SMK DAGs to pull in new data
1. Enable both data refreshes to verify that the data gets pulled and moved without issue
1. (Optional) Run `just down -v && just catalog/pgcli` and verify that the freshly created tables/views now use `text`

## Checklist


      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
